src/expiring_hash_map/tests.rs

```diff
@@ -0,0 +1,92 @@
+use super::*;
+use crate::runtime::Runtime;
+use futures::future::poll_fn;
+use futures_test::task::{new_count_waker, noop_context, panic_context};
+use std::time::{Duration, Instant};
+
+#[test]
+fn poll_does_not_return_ready_with_empty_map() {
+    let mut map = ExpiringHashMap::<String, String>::new();
+    let mut cx = noop_context();
+    assert!(map.poll_expired(&mut cx).is_pending());
+}
+
+#[test]
+fn poll_does_not_return_ready_with_empty_map_after_it_was_non_empty() {
+    let mut rt = Runtime::new().unwrap();
```

Comment:
so for these types of tests we can use this type of type https://docs.rs/tokio-test/0.2.0/tokio_test/task/struct.Spawn.html, happy to chat about testing these things in tokio 0.2 over slack :)

Also the tests dir in the tokio repo is a great place to reference for testing async code.

Reply:
This `Spawn` thing looks very interesting, I'll definitely check it out. It doesn't have usage examples though.